mitosheet/mitosheet/excel_utils.py

```diff
@@ -52,4 +54,69 @@ def get_index_from_column(col: str) -> int:
     for c in col:
         if c in string.ascii_letters:
             num = num * 26 + (ord(c.upper()) - ord('A')) + 1
-    return num - 1
\ No newline at end of file
+    return num - 1
+
+
+def get_table_range_from_upper_left_corner_value(file_path: str, sheet_name: str, value: Union[str, int, float, bool]) -> Optional[str]:
+    """
+    Given a string, this fu ction will look through the excel tab sheet_name at the given
+    file_path, and find the first instance of the value (look through column A first, then B, etc).
+
+    If this value does not exist in the tab, will return None. 
+
+    If the value does exist, then this function will walk down the column until it hits an empty row. Then,
+    it will walk down the final row until it hits an empty column. Then it will return the range that 
+    defines that rectangular of defined data (with value in the upper left corner).
+    """
+    workbook = load_workbook(file_path)
+    sheet = workbook[sheet_name]
+
+    # We get the last defined rows, so we don't waste time searching data we don't need
+    dimension = sheet.calculate_dimension()
+    (min_search_col, min_search_row), (max_search_col, max_search_row) = get_col_and_row_indexes_from_range(dimension)
+    # Unfortunately, openpyxl indexes from 1, so we add one to treat everything in that range
+    (min_search_col, min_search_row), (max_search_col, max_search_row) = (min_search_col + 1, min_search_row + 1), (max_search_col + 1, max_search_row + 1)
+
+    # Loop over the columns one by one to find where this value is set
+    min_found_col_index, min_found_row_index = None, None
+    for col in sheet.iter_cols(min_row=min_search_row, max_row=max_search_row, min_col=min_search_col, max_col=max_search_col):
+        for cell in col:
+            if cell.value == value:
```

Comment:
When looking at frontend, make sure we tell the user that it has to be an exact match with the entire cell value.